tests/pyfunc/test_spark.py

```diff
@@ -124,6 +143,33 @@ def test_spark_udf(spark, model_path):
                 assert expected == actual
 
 
+@pytest.mark.large
+def test_spark_udf_with_conda_env_restored(spark, sklearn_model, model_path):
+    from pyspark.sql.functions import col
+
+    sklearn_log_version = "0.22.1"
+    mlflow.sklearn.save_model(
+        sk_model=sklearn_model.model, path=model_path,
+        pip_requirements=[f"scikit-learn=={sklearn_log_version}"]
+    )
+    infer_data = pd.DataFrame(sklearn_model.inference_data, columns=['a', 'b'])
+
+    infer_spark_df = spark.createDataFrame(infer_data).repartition(1)
+
+    check_dict = {"sklearn": sklearn_log_version}
+    new_env = {
+        **os.environ.copy(),
+        'MLFLOW_SPARK_UDF_RESTORED_ENV_MODULE_VERSION_CHECK_DICT': json.dumps(check_dict)
+    }
```

Comment:
`monkeypatch.setenv` is useful to temporarily set an environment variable:
https://docs.pytest.org/en/6.2.x/monkeypatch.html#monkeypatching-environment-variables